finagle-http/src/main/scala/com/twitter/finagle/http/Request.scala

```diff
@@ -287,10 +287,36 @@ object Request {
    */
   def apply(version: Version, method: Method, uri: String): Request = {
     val reqIn = new DefaultHttpRequest(from(version), from(method), uri)
-    new Request {
-      val httpRequest: HttpRequest = reqIn
-      lazy val remoteSocketAddress: InetSocketAddress = new InetSocketAddress(0)
+    buildRequest(
+      request = reqIn,
+      socketAddress = new InetSocketAddress(0)
+    )
+  }
+
+  private def buildRequest(
+    request: HttpRequest,
+    socketAddress: InetSocketAddress,
+    reader: Option[Reader] = None): Request = {
+
+    val req = reader match {
+      case Some(readerIn: Reader) =>
+        new  Request {
+          override val reader: Reader = readerIn
+          val httpRequest: HttpRequest = request
+          lazy val remoteSocketAddress: InetSocketAddress = socketAddress
+        }
+      case None =>
+        new  Request {
+          val httpRequest: HttpRequest = request
+          lazy val remoteSocketAddress: InetSocketAddress = socketAddress
+        }
+    }
+
+    if (req.version == Version.Http11 && !req.headers.contains(HttpHeaders.Names.HOST)) {
+      req.headers.set(HttpHeaders.Names.HOST, "")
```

Comment:
Thanks for the clarification; this looks correct to me. By adding a link, I meant adding it as a code comment close to this logic in case others are interested in reading the spec in the future.